src/transport/SecureSessionMgr.h

```diff
@@ -92,13 +97,18 @@ class EncryptedPacketBufferHandle final : private System::PacketBufferHandle
     CHIP_ERROR InsertPacketHeader(const PacketHeader & aPacketHeader) { return aPacketHeader.EncodeBeforeData(*this); }
 #endif // CHIP_ENABLE_TEST_ENCRYPTED_BUFFER_API
 
-private:
-    // Allow SecureSessionMgr to assign or construct us from a PacketBufferHandle
-    friend class SecureSessionMgr;
+    void MarkEncrypted(PacketBufferHandle && aBuffer)
```

Comment:
This does both copy and mark.

Either `static EncryptedPacketBufferHandle MarkEncrypted(...)` and then you do:

``cpp
*bufferRetainSlot = EncryptedPacketBufferHandle::MarkEncrypted(msgBuf.Retain());
```

which by the way shows that we have an encrypted buffer that is NOT always marked as such (retain is a copy).

or have `void AssignFromEncrypted(...)` and then you do:

```cpp
bufferRetainSlot->AssignFromEncrypted(msgBuf.Retain());
```

I personally prefer the first version: one can create encrypted handles as long as an actual handle exists.